Jenkinsfile

```diff
@@ -24,36 +48,77 @@ node {
         }
     }
 
-    if (isTag()) {
-        stage('Publish') {
-            nodeEnv.inside("-e HOME=${workspace}") {
-                withCredentials([
-                    [$class: 'StringBinding', credentialsId: 'npm-token', variable: 'NPM_TOKEN']]) {
-
-                    // Use `npm` rather than `yarn` for publishing.
-                    // See https://github.com/yarnpkg/yarn/pull/3391.
-                    sh "echo '//registry.npmjs.org/:_authToken=${env.NPM_TOKEN}' >> \$HOME/.npmrc"
-                    sh "npm publish"
-                }
-            }
+    if (env.BRANCH_NAME != releaseFromBranch) {
+        echo "Skipping deployment because this is not the ${releaseFromBranch} branch"
+        return
+    }
+
+    stage('Publish') {
+        input(message: "Publish new client release?")
 
-            // Upload the contents of the package to an S3 bucket, which it
-            // will then be served from.
-            docker.image('nickstenning/s3-npm-publish')
-                  .withRun('', "hypothesis@${pkgVersion} s3://cdn.hypothes.is") { c ->
-                    sh "docker logs --follow ${c.id}"
-                  }
+        newPkgVersion = bumpMinorVersion(pkgVersion)
+        if (versionSuffix != "") {
+            newPkgVersion = newPkgVersion + "-" + versionSuffix
         }
+
+        echo "Publishing ${pkgName} v${newPkgVersion} from ${releaseFromBranch} branch."
+
+        nodeEnv.inside("-e HOME=${workspace} -e BRANCH_NAME=${env.BRANCH_NAME}") {
+            withCredentials([
+                string(credentialsId: 'npm-token', variable: 'NPM_TOKEN'),
+                usernamePassword(credentialsId: 'github-jenkins-user',
+                                  passwordVariable: 'GITHUB_TOKEN',
+                                  usernameVariable: 'GITHUB_USERNAME')]) {
+
+                // Configure commit author for version bump commit and auth credentials
+                // for pushing tag to GitHub.
+                //
+                // See https://git-scm.com/docs/git-credential-store
+                sh """
+                git config user.email ${env.GITHUB_USERNAME}@hypothes.is
```

Comment:
When I first saw this line, I thought "Oh, if you concatenated my GH username with `@hypothes.is`, you wouldn't get a functional email address." But then I realized that `GITHUB_USERNAME` is de facto always the H Jenkins GH user. So this line is funky to me because it's sort of half static. But, eh, importance level? Tiny.

Reply:
The email address isn't actually used for anything here. It's just a placeholder.